extensions/ql-vscode/test/vscode-tests/no-workspace/databases/local-databases-ui.test.ts

```diff
@@ -121,6 +120,106 @@ describe("local-databases-ui", () => {
     databaseUI.dispose(testDisposeHandler);
   });
 
+  describe("getDatabaseItem", () => {
+    const progress = jest.fn();
+    const token = new CancellationTokenSource().token;
+    it("should return current database if there is one", async () => {
+      const databaseUI = new DatabaseUI(
+        app,
+        {
+          databaseItems: [{ databaseUri: Uri.file(db1) }],
+          onDidChangeDatabaseItem: () => {
+            /**/
+          },
+          onDidChangeCurrentDatabaseItem: () => {
+            /**/
+          },
+          setCurrentDatabaseItem: () => {},
+          currentDatabaseItem: { databaseUri: Uri.file(db1) },
+        } as any,
+        {
+          onLanguageContextChanged: () => {
+            /**/
+          },
+        } as any,
+        {} as any,
+        storageDir,
+        storageDir,
+      );
+
+      const databaseItem = await databaseUI.getDatabaseItem(progress, token);
+
+      expect(databaseItem).toBeDefined();
+    });
+
+    it("should prompt for a database if there is no current one", async () => {
+      const databaseUI = new DatabaseUI(
+        app,
+        {
+          databaseItems: [{ databaseUri: Uri.file(db1) }],
+          onDidChangeDatabaseItem: () => {
+            /**/
+          },
+          onDidChangeCurrentDatabaseItem: () => {
+            /**/
+          },
+          setCurrentDatabaseItem: () => {},
+          currentDatabaseItem: undefined,
+        } as any,
+        {
+          onLanguageContextChanged: () => {
+            /**/
+          },
+        } as any,
+        {} as any,
+        storageDir,
+        storageDir,
+      );
+
+      const showQuickPickSpy = jest
+        .spyOn(window, "showQuickPick")
+        .mockResolvedValue(
+          mockedQuickPickItem(
+            mockedObject<DatabaseSelectionQuickPickItem>({
+              databaseKind: "existing",
+            }),
+          ),
+        );
+
+      const selectExistingDatabaseSpy = jest
+        .spyOn(databaseUI as any, "selectExistingDatabase")
+        .mockResolvedValue(undefined);
+
+      const importNewDatabaseSpy = jest
+        .spyOn(databaseUI as any, "importNewDatabase")
+        .mockResolvedValue(undefined);
```

Comment:
Slightly tweaked the tests so didn't have to make all the methods public 🤫 This meant I had to use `as any` here to spy on these methods, but that's probably better?

Reply:
Hmm so we're mocking out methods on the object we're testing? That seems a bit odd. Doesn't it means that we're not actually testing them?

Reply:
Yeah, I haven't added tests for `selectExistingDatabase` and `importNewDatabase` in this PR, just for the overall `promptForDatabase` dropdown 👀  

Happy to have a go now if that makes more sense? Though I was also planning to open a follow-up issue to improve the feature + tests generally 💅🏽

Reply:
It's a little bit odd to me that we're testing a function that calls other functions but those other functions are being mocked. You'd normally mock out dependencies etc. but have real implementations of the component that is being tested.

But it's a step forward so I'm happy for us to do some further improvements later on.